src/enigma2/data/Tags.h

```diff
@@ -48,7 +53,7 @@ namespace enigma2
         return (regex_match(m_tags, regex));
       }
 
-      void AddTag(const std::string &tagName, const std::string &tagValue = "")
+      void AddTag(const std::string &tagName, std::string tagValue = "", bool replaceUnderscores = false)
```

Comment:
why the change from const std::string& to `by value` parameter?

Reply:
Because of the default value.

Reply:
imo a default value `const std::string& foo = ""`does work just fine.

Reply:
So it does ;)

I passed by value instead of creating a new variable in the method to be used here:

`std::replace(tagValue.begin(), tagValue.end(), ' ', '_');`

Is it better to pass const and make a copy?

Reply:
Your decision. Doesn't really make a difference, I'd say.

Reply:
I'd go with the const ref way + copy in function body. So, it is very clear what happens. Performance-wise it does not make a difference, as both approaches need one copy.

Reply:
Ok, cool will push an update for that later today.

Reply:
Your decision. From my pow the PR can go in as it is now.

Reply:
Then in it goes...